gt_concordance/gt_concordance.py

```diff
@@ -0,0 +1,514 @@
+    if SNPS_FILE is None:
+        click.echo('config file missing SNPS_FILE')
+        exit(0)
+    if not os.path.exists(SNPS_FILE):
+        click.echo('SNPS file does not exist! Path: {}'.format(SNPS_FILE))
+        exit(0)
+
+    xl_files = [os.path.join(XL_FILES_PATH, filename) for filename in os.listdir(XL_FILES_PATH) if '.xlsx' in filename]
+    if not xl_files:
+        click.echo('No .xlsx files found! Terminating')
+        exit(0)
+
+    return True
+
+def parse_maf_snps_file(config):
+    SNPS_FILE = config.get('SNPS_FILE')
+    snps_data = {}
+    if os.path.exists(SNPS_FILE):
+        with open(SNPS_FILE) as snps_file:
+            lines = snps_file.readlines()
+            for line in lines:
+                chromosome, position, rs_position, reference, alternative = line.split()
+                snps_data[rs_position] = [chromosome, position, rs_position, reference, alternative]
+    return snps_data
+
+
+@cli.command()
+@click.argument('sample')
+@click.option('--force', '-f', is_flag=True, default=False, help='If not specified, will keep existing vcf files and use them to check concordance. Otherwise overwrite')
+@click.pass_context
+def genotype_sample(context, sample, force):
+    if is_config_file_ok():
+        concordance = run_genotype_sample(sample, force)
+        click.echo('Sample name\t % concordance')
+        click.echo('{}:\t {}'.format(sample, concordance))
+
+@click.pass_context
+def run_genotype_sample(context, sample, force=None):
+    config = context.obj
+    project = sample.split('_')[0]
+    output_path = os.path.join(config.get('ANALYSIS_PATH'), project, 'piper_ngi/03_genotype_concordance')
+
+    # check if gt file exists
+    gt_file = os.path.join(output_path, '{}.gt'.format(sample))
+    if not os.path.exists(gt_file):
+        click.echo('gt file does not exist! Path: {}'.format(gt_file))
+        click.echo('To create .gt file run the command: gt_concordance parse_xl_files')
+        exit(0)
+
+    # if we are here, the path has been already checked (most *likely*)
+    if os.path.exists(output_path):
+        # check if gatk needs to be run
+        vcf_file = os.path.join(output_path, "{}.vcf".format(sample))
+        to_run_gatk = False
+        if os.path.exists(vcf_file):
+            click.echo('.vcf file already exists: {}'.format(os.path.basename(vcf_file)))
+            if force:
+                click.echo('Rerunning GATK because of --force option')
+                to_run_gatk = True
+            else:
+                click.echo('Skipping GATK')
+        else:
+            click.echo('Running GATK')
+            to_run_gatk = True
+
+        # run gatk if needed
+        if to_run_gatk:
+            vcf_file = run_gatk(sample, config)
+            if vcf_file is None:
+                click.echo('GATK completed with ERROR!')
+                click.echo('Terminating')
+                # update charon
+                update_charon(sample=sample, status='FAILED')
+                exit(0)
```

Comment:
there is no need to terminate the entire program, anyway you are capturing the fact that something failed so you can ho on with the next sample